tests/test_modeling_tf_wav2vec2.py

```diff
@@ -0,0 +1,349 @@
+            pt_output = pt_layer(torch.from_numpy(test))[0].detach().numpy()
+
+            tf.testing.assert_near(tf_output, pt_output, atol=1e-3)
+
+    def check_feature_projector(self):
+        tf_layer = self.tf_model.wav2vec2.feature_projection
+        pt_layer = self.pt_model.wav2vec2.feature_projection
+
+        test = np.random.rand(1, 292, 768).astype(np.float32)
+
+        tf_input = test
+        pt_input = torch.from_numpy(test)
+
+        tf_output = tf_layer(tf_input).last_hidden_state.numpy()
+        pt_output = pt_layer(pt_input).last_hidden_state.detach().numpy()
+
+        tf.testing.assert_near(tf_output, pt_output, atol=1e-3)
+
+    def check_feature_extractor(self):
+        tf_layer = self.tf_model.wav2vec2.feature_projection
+        pt_layer = self.pt_model.wav2vec2.feature_projection
+
+        test = np.random.rand(1, 16000).astype(np.float32)
+
+        tf_input = test
+        pt_input = torch.from_numpy(test)
+
+        tf_output = tf_layer(tf_input).last_hidden_state.numpy()
+        pt_output = pt_layer(pt_input).last_hidden_state.detach().numpy()
+
+        tf.testing.assert_near(tf_output, pt_output, atol=1e-3)
+
+    def check_feed_forward(self):
+        config = Wav2Vec2Config()
+        for i in range(config.num_hidden_layers):
+            tf_layer = self.tf_model.wav2vec2.encoder.layer[i].feed_forward
+            pt_layer = self.pt_model.wav2vec2.encoder.layers[i].feed_forward
+
+            test = np.random.rand(1, 292, 768).astype(np.float32)
+
+            tf_input = test
+            pt_input = torch.from_numpy(test)
+
+            tf_output = tf_layer(tf_input)[0].numpy()
+            pt_output = pt_layer(pt_input)[0].detach().numpy()
+
+            tf.testing.assert_near(tf_output, pt_output, atol=1e-3)
+
+    def check_ctc_head(self):
+        tf_layer = self.tf_model.lm_head
+        pt_layer = self.pt_model.lm_head
+
+        test = np.random.rand(1, 292, 768).astype(np.float32)
+
+        tf_input = test
+        pt_input = torch.from_numpy(test)
+
+        tf_output = tf_layer(tf_input).numpy()
+        pt_output = pt_layer(pt_input).detach().numpy()
+
+        tf.testing.assert_near(tf_output, pt_output, atol=1e-3)
+
+    def check_encoder(self):
+        tf_layer = self.tf_model.wav2vec2.encoder
+        pt_layer = self.pt_model.wav2vec2.encoder
+
+        test = np.random.rand(1, 292, 768).astype(np.float32)
+
+        tf_input = test
+        pt_input = torch.from_numpy(test)
+
+        tf_output = tf_layer(tf_input).last_hidden_state.numpy()
+        pt_output = pt_layer(pt_input).last_hidden_state.detach().numpy()
+
+        tf.testing.assert_near(tf_output, pt_output, atol=1e-3)
+
+    def create_and_check_batch_inference(self, config, input_values, *args):
+        # test does not pass for models making use of `group_norm`
+        # check: https://github.com/pytorch/fairseq/issues/3227
+        pass
+
+    def prepare_config_and_inputs_for_common(self):
```

Comment:
could we try to translate the tests from PyTorch to Tensorflow: https://github.com/huggingface/transformers/blob/cfca638acbc9361fca43a6fcff33fa512ab5df93/tests/test_modeling_wav2vec2.py#L122

instead of comparing TF<=>PT? This way we can be use that the model works independently of changes to the PyTorch model.